### PR TITLE
add cache metrics

### PR DESCRIPTION
**Pull request by matsumana**
Hello.

With reference to #87, I added metrics the following.

Please review.

```
# HELP elasticsearch_indices_query_cache_cache_size Query cache cache size
# TYPE elasticsearch_indices_query_cache_cache_size gauge
elasticsearch_indices_query_cache_cache_size{cluster="test"} 28869
# HELP elasticsearch_indices_query_cache_cache_count Query cache cache count
# TYPE elasticsearch_indices_query_cache_cache_count gauge
elasticsearch_indices_query_cache_cache_count{cluster="test"} 28925
# HELP elasticsearch_indices_query_cache_total Query cache total count
# TYPE elasticsearch_indices_query_cache_total counter
elasticsearch_indices_query_cache_total{cluster="test"} 4.916801e+06
# HELP elasticsearch_indices_query_cache_count Query cache count
# TYPE elasticsearch_indices_query_cache_count counter
elasticsearch_indices_query_cache_count{category="hit",cluster="test"} 1.958521e+06
elasticsearch_indices_query_cache_count{category="miss",cluster="test"} 2.95828e+06
# HELP elasticsearch_indices_request_cache_count Request cache count
# TYPE elasticsearch_indices_request_cache_count counter
elasticsearch_indices_request_cache_count{category="hit",cluster="test"} 75749
elasticsearch_indices_request_cache_count{category="miss",cluster="test"} 65144
```

### CODE REVIEW

File: collector/nodes.go [hunk 1]
```diff
@@ -1038,6 +1128,22 @@ func (c *Nodes) Collect(ch chan<- prometheus.Metric) {
 				metric.Labels(nodeStatsResponse.ClusterName, node)...,
 			)
 		}
+		for _, metric := range c.nodeMetricsVec {
+			labelValues := metric.LabelValues(nodeStatsResponse.ClusterName, node)
+			values := metric.Values(node)
+
+			for i, labelValue := range labelValues {
+				counterVec := prometheus.NewCounterVec(prometheus.CounterOpts{
```
Comment (zwopir) [inline collector/nodes.go]: `NewCounterVec` must not be called in the `Collect` func implementing `prometheus.Collector`. Throughout the collector we are using the pattern to create a new const metric. (`prometheus.MustNewConstMetric`). It perfectly fits to our use case.

File: collector/nodes.go [hunk 2]
```diff
@@ -1038,6 +1128,22 @@ func (c *Nodes) Collect(ch chan<- prometheus.Metric) {
...
+					Namespace: namespace,
+					Subsystem: metric.Subsystem,
+					Name:      metric.Name,
+					Help:      metric.Help},
+					metric.Labels)
+				counterVec.WithLabelValues(labelValue...).Set(values[i])
+				counterVec.Collect(ch)
+			}
+
```
Comment (zwopir) [inline collector/nodes.go]: this metric returns false results regarding the labels: on our cluster we get: 
```
# TYPE elasticsearch_indices_query_cache_count counter
elasticsearch_indices_query_cache_count{category="hit",cluster="cluster",host="host",name="name"} 3.1548049e+07
elasticsearch_indices_query_cache_count{category="miss",cluster="cluster",host="host",name="name"} 4.9529376e+07
```
(the label values for host and name are not (!) redacted)

**Comment by matsumana** [comment]
@zwopir Thank you for the comments.
I fixed.
Would you please review again?

**Comment by zwopir** [comment]
yes, that looks good - Thanks again for your contribution!

**Pull request merged**